apps/web/src/components/WalletConnect/WalletConnectProvider.tsx

```diff
@@ -0,0 +1,119 @@
+import { type NetworkType } from "@airgap/beacon-wallet";
+import { useToast } from "@chakra-ui/react";
+import { type WalletKitTypes } from "@reown/walletkit";
+import { useDynamicModalContext } from "@umami/components";
+import {
+  createWalletKit,
+  useAsyncActionHandler,
+  useAvailableNetworks,
+  useWcPeers,
+  walletKit,
+} from "@umami/state";
+import { type Network } from "@umami/tezos";
+import { formatJsonRpcError } from "@walletconnect/jsonrpc-utils";
+import { getSdkError } from "@walletconnect/utils";
+import { type PropsWithChildren, useEffect } from "react";
+
+import { SessionProposalModal } from "./SessionProposalModal";
+
+export const context = {};
+
+export const WalletConnectProvider = ({ children }: PropsWithChildren) => {
+  const onSessionProposal = useOnSessionProposal();
+  const onSessionDelete = useOnSessionDelete();
+  const onSessionRequest = useOnSessionRequest();
+
+  useEffect(() => {
+    const initializeWallet = async () => {
+      await createWalletKit();
+      walletKit.on("session_proposal", event => void onSessionProposal(event));
+      walletKit.on("session_request", event => void onSessionRequest(event));
+      walletKit.on("session_delete", event => void onSessionDelete(event));
+    };
+
+    void initializeWallet();
+  });
```

Comment:
Do we need to call it on each render? 
I think we should init wallet only on page mounting, so please set deps array like

` useEffect(() => {
....
  }, []);`
  
 you might want to read [this](https://react.dev/learn/synchronizing-with-effects#step-2-specify-the-effect-dependencies) to get a better understanding of how the effects work in react